prerequisites.py

```diff
@@ -0,0 +1,411 @@
+        print("  %s" % country_code)
+        path = os.path.join("cms", "server", "mo", country_code, "LC_MESSAGES")
+        compiled_path = os.path.join(path, "cms.mo")
+        if not os.path.exists(compiled_path):
+            print("[Error] %s not found" % (compiled_path))
+            print("[Error] You must run the compile_l10n command")
+            exit(1)
+        elif os.path.getmtime(locale) > os.path.getmtime(compiled_path):
+            print("[Warning] %s is newer than %s" % (locale, compiled_path))
+            print("[Warning] Are you sure you ran the compile_l10n command?")
+
+    for locale in glob(os.path.join("cms", "server", "po", "*.po")):
+        country_code = re.search(r"/([^/]*)\.po", locale).groups()[0]
+        print("  %s" % country_code)
+        path = os.path.join("cms", "server", "mo", country_code, "LC_MESSAGES")
+        dest_path = os.path.join(USR_ROOT, "share", "locale",
+                                 country_code, "LC_MESSAGES")
+        makedir(dest_path, root, 0755)
+        copyfile(os.path.join(path, "cms.mo"),
+                 os.path.join(dest_path, "cms.mo"),
+                 root, 0644)
+
+
+def compile_isolate():
+    """This function compiles the isolate sandbox.
+
+    """
+
+    check_root(False)
+
+    print("===== Compiling isolate")
+    os.chdir("isolate")
+    os.system("make")
+    os.chdir("..")
+
+
+def install_isolate():
+    """This function installs the isolate sandbox.
+
+    """
+
+    check_root(True)
+    root = pwd.getpwnam("root")
+    try:
+        cmsuser = pwd.getpwnam("cmsuser")
+        cmsuser_grp = grp.getgrnam("cmsuser")
+    except:
+        print("[Error] The cmsuser doesn't exist yet")
+        print("[Error] You need to run the install command at least once")
+        exit(1)
+
+    print("===== Copying isolate to /usr/local/bin/")
+
+    # Check if compile_isolate() has been called
+    if not os.path.exists(os.path.join("isolate", "isolate")):
+        print("[Error] You must run the compile_isolate command first")
+        exit(1)
+
+    makedir(os.path.join(USR_ROOT, "bin"), root, 0755)
+    copyfile(os.path.join(".", "isolate", "isolate"),
+             os.path.join(USR_ROOT, "bin", "isolate"),
+             root, 04750, group=cmsuser_grp)
+
+
+def install_all():
+    """This function prepares all that's needed to run CMS:
+    - creation of cmsuser user
+    - installation of isolate
+    - installation of localization files
+    - installation of configuration files
+    and so on.
+
+    """
+
+    check_root(True)
+
+    print("===== Creating user and group cmsuser")
+    os.system("useradd cmsuser -c 'CMS default user' -M -r -s /bin/false -U")
+    cmsuser = pwd.getpwnam("cmsuser")
+    root = pwd.getpwnam("root")
+    cmsuser_grp = grp.getgrnam("cmsuser")
+
+    # Run compile_l10n() and compile_isolate() as *not* root
+    if os.system("sudo -u %s %s compile_l10n" % (os.getenv("SUDO_USER"), sys.argv[0])):
```

Comment:
IIUC, SUDO_USER is set when you run something as sudo. The fact that we are root doesn't mean we are running through sudo, though, so you might want to check that this is set before using it.